mllib/src/main/scala/org/apache/spark/mllib/tree/impurity/Impurity.scala

```diff
@@ -90,6 +90,11 @@ private[spark] abstract class ImpurityAggregator(val statsSize: Int) extends Ser
    */
   def getCalculator(allStats: Array[Double], offset: Int): ImpurityCalculator
 
+  /**
+   * Get an [[ImpurityCalculator]] for a node.
+   * @param sufficientStats  Sufficient stats array for a node.
+   */
+  def getCalculator(sufficientStats: Array[Double]): ImpurityCalculator
```

Comment:
You should be able to use the existing getCalculator method, passing in offset 0.